trajopt_ifopt/include/trajopt_ifopt/constraints/inverse_kinematics_constraint.h

```diff
@@ -0,0 +1,180 @@
+ *
+ * @author Matthew Powelson
+ * @date May 18, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TRAJOPT_IFOPT_INVERSE_KINEMATICS_CONSTRAINT_H
+#define TRAJOPT_IFOPT_INVERSE_KINEMATICS_CONSTRAINT_H
+
+#include <trajopt_utils/macros.h>
+TRAJOPT_IGNORE_WARNINGS_PUSH
+#include <Eigen/Eigen>
+#include <ifopt/constraint_set.h>
+
+#include <tesseract_kinematics/core/inverse_kinematics.h>
+#include <tesseract_environment/core/environment.h>
+#include <tesseract_environment/core/utils.h>
+TRAJOPT_IGNORE_WARNINGS_POP
+
+#include <trajopt/utils.hpp>
+#include <trajopt_ifopt/variable_sets/joint_position_variable.h>
+
+namespace trajopt
+{
+/**
+ * @brief Contains kinematic information for the inverse kinematics constraint
+ */
+struct InverseKinematicsInfo
+{
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  using Ptr = std::shared_ptr<InverseKinematicsInfo>;
+  using ConstPtr = std::shared_ptr<const InverseKinematicsInfo>;
+
+  InverseKinematicsInfo() = default;
+  InverseKinematicsInfo(tesseract_kinematics::InverseKinematics::ConstPtr inverse_kinematics,
+                        tesseract_environment::AdjacencyMap::ConstPtr adjacency_map,
+                        const Eigen::Isometry3d& world_to_base,
+                        std::string link,
+                        const Eigen::Isometry3d& tcp = Eigen::Isometry3d::Identity())
+    : inverse_kinematics_(std::move(inverse_kinematics))
+    , adjacency_map_(std::move(adjacency_map))
+    , world_to_base_(world_to_base)
+    , link_(std::move(link))
+    , tcp_(tcp)
+  {
+    kin_link_ = adjacency_map_->getLinkMapping(link_);
+    if (kin_link_ == nullptr)
+    {
+      CONSOLE_BRIDGE_logError("Link name '%s' provided does not exist.", link_.c_str());
+      assert(false);
+    }
+  }
+
+  tesseract_kinematics::InverseKinematics::ConstPtr inverse_kinematics_;
+  /** @brief Not currently respected */
+  tesseract_environment::AdjacencyMap::ConstPtr adjacency_map_;
+  /** @brief Not currently respected */
+  Eigen::Isometry3d world_to_base_;
+  /** @brief Not currently respected */
+  std::string link_;
+  /** @brief Not currently respected */
+  tesseract_environment::AdjacencyMapPair::ConstPtr kin_link_;
+  /** @brief Not currently respected */
+  Eigen::Isometry3d tcp_;
```

Comment:
Remove trailing underscore on public member variables.

Reply:
I have done this. I should note that the members are now named the same thing as the constructor arguments. Is that bad practice? Also, I will note that the ROS style guide does not dictate that public members don't have this. I know that is the style we are using in Tesseract, but I just thought I would mention this.

> Member variables
Variables that are members of a class (sometimes called fields) are under_scored, with a trailing underscore added.


http://wiki.ros.org/CppStyleGuide#Member_variables

Reply:
Yea just to be consistent.

Reply:
I forgot, I believe the ROS style guide is based on google style guide which has this distinction between struct and classes.

https://google.github.io/styleguide/cppguide.html#Variable_Names